tests/integration/tester.py

```diff
@@ -0,0 +1,46 @@
+# Copyright 2024 Canonical Ltd.
+# See LICENSE file for licensing details.
+
+import textwrap
+
+ANY_CHARM = textwrap.dedent(
```

Comment:
What's the reason for having this instead of a minimal but real charm stored under the `integration` dir? This seems pretty complicated. Charm code is here, overwrites in the `test_charm.py`, charm name in the `conftest.py`.